common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -457,18 +457,19 @@ static ConcurrentMap<AppExecId, ExecutorShuffleInfo> reloadRegisteredExecutors(D
       throws IOException {
     ConcurrentMap<AppExecId, ExecutorShuffleInfo> registeredExecutors = Maps.newConcurrentMap();
     if (db != null) {
-      DBIterator itr = db.iterator();
```

Comment:
`reloadActiveAppAttemptsPathInfo` and `reloadFinalizedAppAttemptsShuffleMergeInfo` should be Spark 3.4 only. Please let me know if this PR needs to be backport to other branch